python/tvm/contrib/cutlass/build.py

```diff
@@ -862,10 +862,13 @@ def handle_matmul(self, f, op_type):
     def handle_attention(self, f, op_type):
         """Annotate an attention op."""
         signature = _extract_relax_function_signature(f)
+
         if _get_call_node(f.body, "relax.nn.attention") is not None:
             op_attrs = _get_call_node(f.body, "relax.nn.attention").attrs
         elif _get_call_node(f.body, "relax.nn.attention_bias") is not None:
             op_attrs = _get_call_node(f.body, "relax.nn.attention_bias").attrs
+        elif _get_call_node(f.body, "relax.nn.attention_var_len") is not None:
```

Comment:
I am wondering if there is flexibility in Relax ops system to unify all these attention variations. The current approach of having a different op name for each variation is okay, but doesn't seem to be scalable if we want to introduce more variations (like var_len + bias, customized mask)

Reply:
yeah, it would be nice if we can define an "attention input" object to pass all inputs, rather than individually.

Reply:
Or another alternative could be having a single attention with a superset of attributes for different variants of attention. Based on how the attributes are set, we may be able to differentiate different attentions.

Reply:
That doesn't work because different kinds of attention op require different inputs, not attributes. And Relax doesn't support variable-length inputs.